File: frontend/resize.py
from PIL import Image

img = Image.open('icon.png')
size = max(img.size)
new_img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
new_img.paste(img, ((size - img.size[0]) // 2, (size - img.size[1]) // 2))
new_img = new_img.resize((256, 256), Image.Resampling.LANCZOS)
new_img.save('icon_square.png')

File: frontend/resize.js
import { Jimp } from 'jimp';

Jimp.read('icon.png').then(img => {
  const size = Math.max(img.bitmap.width, img.bitmap.height);
  const bg = new Jimp({ width: size, height: size, color: 0x00000000 });
  bg.composite(img, Math.floor((size - img.bitmap.width) / 2), Math.floor((size - img.bitmap.height) / 2));
  bg.resize({ w: 256, h: 256 });
  bg.write('icon_square.png');
  console.log('Done');
}).catch(console.error);
